### ISSUE TITLE
blktrace no longer monitors actions with latest 0.6.5

### ISSUE THREAD

**Issue by koplover** [zfsonlinux/zfs#4868]
I've been using 0.6.4.2 ZFS for some time, and used blktrace to following the block level actions on specific zvols, useful to watch the whole lifecycle of each request.

I run various VMs under Xen hypervisor, with the zvols providing the VM disks. However, since moving to 0.6.5.7 (and I've tried various earlier versions in 0.6.5 stream. although believe it worked for 0.6.5.2) I now see only Q (Queued) actions.

So the blktrace/blkparse output now very un-usefully looks like the below:

230,112  2        9     0.000755797 11721  Q   R 385152 + 128 [blkback.27.hdd]
230,112  2       10     0.036427203 11721  Q   R 101991040 + 128 [blkback.27.hdd]
230,112  2       11     0.036705229 11721  Q   R 385280 + 128 [blkback.27.hdd]
230,112  2       12     0.036758867 11721  Q   R 385408 + 128 [blkback.27.hdd]
230,112  2       13     0.187958398 11721  Q   R 385536 + 128 [blkback.27.hdd]
230,112  2       14     0.188025318 11721  Q   R 385664 + 128 [blkback.27.hdd]
230,112  2       15     0.188074096 11721  Q   R 385792 + 128 [blkback.27.hdd]
230,112  2       16     0.188125051 11721  Q   R 385920 + 128 [blkback.27.hdd]
230,112  2       17     0.188176186 11721  Q   R 386048 + 128 [blkback.27.hdd]
230,112  2       18     0.188231052 11721  Q   R 386176 + 128 [blkback.27.hdd]

Is this a known regression? Is there anything known to re-enable the block level tracing?

**Comment by behlendorf** [comment]
This may have been an side effect of 37f9dac592bf5889c3efb305c48ac39b4c7dd140 which was merged for 0.6.5.x and doesn't appear in 0.6.4.x.

**Comment by koplover** [comment]
@behlendorf yes this seems to be the case. 

I've deployed a version of 0.6.4.2 which has this commit backported (in fact I cheated and retrieved the merge created by ryao in issue 3888, which merged back 4 commits but this was the significant one to port back performance improvements)

blktrace again only shows the Q actions as for 0.6.5, so it would point very much at that commit. The commit is quite large and been around for some time, so removing it from the 0.6.5 code for a quick test looks a little difficult.

Any workarounds that can be used to let me get at the actions. The comment in the commit around loop devices won't help in this instance obviously as the block actions will not be initiated from there?
